src/Elmish.WPF/ViewModel.fs

```diff
@@ -315,7 +316,7 @@ and internal Initialize
           let chain = LoggingViewModelArgs.getNameChainFor nameChain name
           d.GetModel initialModel
           |> ValueOption.map (fun m -> ViewModelArgs.create m (toMsg >> dispatch) chain loggingArgs)
-          |> ValueOption.map (fun args -> ViewModel(args, d.GetBindings ()))
```

Comment:
Right here, if you look at the old way of doing it, the old `ViewModel` constructor was always called with `obj` arguments. But now, we call `CreateViewModel` which unboxes/boxes the values before passing them into the constructor.